snapshot/src/main/java/io/zeebe/snapshots/broker/impl/FileBasedSnapshotStore.java

```diff
@@ -388,6 +406,23 @@ PersistedSnapshot newSnapshot(final FileBasedSnapshotMetadata metadata, final Pa
     return newPersistedSnapshot;
   }
 
+  private Path buildChecksumFilePath(final FileBasedSnapshotMetadata metadata) {
+    return snapshotsDirectory.resolve(metadata.getSnapshotIdAsString() + CHECKSUM_SUFFIX);
+  }
+
+  private void writeChecksumFile(final Path checksumFile, final long checksum) {
+    try {
+      SnapshotChecksum.persist(checksumFile, checksum);
+    } catch (final IOException e) {
```

Comment:
If this fails, shouldn't we delete the snapshot directory?

Reply:
Not if we write the checksum file first :stuck_out_tongue:

Reply:
Yes :+1: